test/parallel/test-http-agent-abort-controller.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+const Agent = require('_http_agent').Agent;
+const { getEventListeners } = require('events');
+const agent = new Agent();
+const server = http.createServer();
+
+server.listen(0, common.mustCall(() => {
+  const port = server.address().port;
+  const host = 'localhost';
+  const options = {
+    port: port,
+    host: host,
+    rejectUnauthorized: false,
+    _agentKey: agent.getName({ port, host })
+  };
+
+  function postCreateConnection() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      const connection = agent.createConnection({ ...options, signal });
+      connection.on('error', common.mustCall((err) => {
+        assert(err);
+        assert.strictEqual(err.name, 'AbortError');
+        res();
+      }));
+      assert.strictEqual(getEventListeners(signal, 'abort').length, 1);
+      ac.abort();
+    });
+  }
+
+  function preCreateConnection() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      ac.abort();
+      const connection = agent.createConnection({ ...options, signal });
+      connection.on('error', common.mustCall((err) => {
+        assert(err);
+        assert.strictEqual(err.name, 'AbortError');
+        res();
+      }));
+    });
+  }
+
+  function agentAsParam() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      const request = http.get({
+        port: server.address().port,
+        path: '/hello',
+        agent: agent,
+        signal,
+      });
+      request.on('error', common.mustCall((err) => {
+        assert(err);
+        assert.strictEqual(err.name, 'AbortError');
+        res();
+      }));
+      assert.strictEqual(getEventListeners(signal, 'abort').length, 1);
+      ac.abort();
+    });
+  }
+
+  function agentAsParamPreAbort() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      ac.abort();
+      const request = http.get({
+        port: server.address().port,
+        path: '/hello',
+        agent: agent,
+        signal,
+      });
+      request.on('error', common.mustCall((err) => {
+        assert(err);
+        assert.strictEqual(err.name, 'AbortError');
+        res();
+      }));
+      assert.strictEqual(getEventListeners(signal, 'abort').length, 0);
+    });
+  }
+
+  postCreateConnection()
+    .then(common.mustCall(preCreateConnection))
+    .then(common.mustCall(agentAsParam))
+    .then(common.mustCall(agentAsParamPreAbort))
```

Comment:
nit: no need for several `mustCall` calls in the same promise chain
```suggestion
    .then(preCreateConnection)
    .then(agentAsParam)
    .then(agentAsParamPreAbort)
```